scripts/src/precheck/submission.py

```diff
@@ -0,0 +1,220 @@
+    source: Source = field(default_factory=lambda: Source())
+    tarball: Tarball = field(default_factory=lambda: Tarball())
+
+    def __post_init__(self):
+        """Complete the initialization of the Submission object.
+
+        modified_files should be gatered and validated at the beginning of the pipeline, as part
+        of the "pre-check" job. Hence this isn't done when a Submission object is loaded from a
+        JSON file aka when modified_files is not empty.
+        """
+        if not self.modified_files:
+            self.modified_files = []
+            self._get_modified_files()
+            self._parse_modified_files()
+
+    def _get_modified_files(self):
+        """Query the GitHub API in order to retrieve the list of files that are added / modified by this PR"""
+        page_number = 1
+        max_page_size, page_size = 100, 100
+        headers = {
+            "Accept": "application/vnd.github.v3+json",
+            "Authorization": f'Bearer {os.environ.get("BOT_TOKEN")}',
+        }
+        files_api_url = f"{self.api_url}/files"
+
+        while page_size == max_page_size:
+            files_api_query = f"{files_api_url}?per_page={page_size}&page={page_number}"
+            print(f"[INFO] Query files : {files_api_query}")
+            r = requests.get(files_api_query, headers=headers)
+            files = r.json()
+            page_size = len(files)
+            page_number += 1
+
+            if xRateLimit in r.headers:
+                print(f"[DEBUG] {xRateLimit} : {r.headers[xRateLimit]}")
+            if xRateRemain in r.headers:
+                print(f"[DEBUG] {xRateRemain}  : {r.headers[xRateRemain]}")
+
+            if "message" in files:
+                msg = f'[ERROR] getting pr files: {files["message"]}'
+                print(msg)
+                raise SubmissionError(msg)
+            else:
+                for file in files:
+                    if "filename" in file:
+                        self.modified_files.append(file["filename"])
+
+    def _parse_modified_files(self):
+        """Classify the list of modified files.
+
+        This method will populate the chart, report, source and tarball attributes.
+
+        We expect the user to provide either:
+        * Only a report file
+        * Only a chart - either as source or tarball (potentially accompagnied by provenance file)
+        * Both the report and the chart
+
+        Raises a SubmissionError if:
+        * The Submission concerns more than one chart
+        * The version of the chart is not SemVer compatible
+        * The tarball file is names incorrectly
+        * The PR contains additional files, not related to the Chart being submitted
+        * The user attempts to create the OWNERS file for its project.
+
+        """
+        pattern, reportpattern, tarballpattern = (
+            checkpr.get_file_match_compiled_patterns()
+        )
+        matches_found = 0
+        none_chart_files = {}
```

Comment:
NIT: Rename this (and all references).

```suggestion
        non_chart_files = {}
```

I know this is a carryover from other code. If you want to keep it the way it is for this PR to reduce the diff surface area, that's fine.

Reply:
Good catch, I didn't even realized this typo in the original code ! I will change it.

FYI I'm not happy with the current implementation, especially of the `_parse_modified_files` method, but I needed something that works so I stole the code from checkpr. I would like to revisit it before getting this merged ideally (more to come in a further commit).

My intention was to discuss the general design before spending too much time refining the implementation.

In other word, I don't think the "diff surface area" is going to be small anyway and don't mind taking the time to rewrite part of the existing script, rather than trying to use the exact same code as much as possible.

Reply:
> FYI I'm not happy with the current implementation, especially of the _parse_modified_files method, but I needed something that works so I stole the code from checkpr. I would like to revisit it before getting this merged ideally (more to come in a further commit).

Totally understand. 100% agree. It could use a clarity rewrite, but it works so I haven't prioritized it.